src/main/java/io/jenkins/plugins/coverage/CoveragePublisher.java

```diff
@@ -92,12 +92,18 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnul
         processor.setFailNoReports(failNoReports);
         processor.setApplyThresholdRecursively(applyThresholdRecursively);
 
+        CoverageAction action = null;
         try {
-            processor.performCoverageReport(reportAdapters, reportDetectors, globalThresholds);
+            action = processor.performCoverageReport(reportAdapters, reportDetectors, globalThresholds);
         } catch (CoverageException e) {
             listener.getLogger().println(ExceptionUtils.getFullStackTrace(e));
             run.setResult(Result.FAILURE);
         }
+
+        if (action != null) {
```

Comment:
Move inside the `try` then `action = null` and the `if` is not required at all.

Reply:
I just tried to keep the `try-catch` section as small as possible for easier exception locating, so isn't this a concern?